src/tool-core/models/tool-core/translation_utility_functions/translation_utility_functions.xtuml

```diff
@@ -238,11 +241,19 @@ INSERT INTO S_SYNC
 	"00000000-0000-0000-0000-000000000000",
 	'addTypeToImports',
 	'',
-	'type_to_import = param.type_to_import;
-
+	'file = param.file;
+imptype = param.type;
+type_to_import = param.type_to_import;
 if ( not type_to_import.primitive() and "" != type_to_import.package )
-  ::addStringToImports( file:param.file, s:type_to_import.package + "." + type_to_import.name, type:param.type );
-end if;',
+  import_req = ImportRequest::get_type_import( type:type_to_import );
+  import_reference = ImportReference::get( file:file, request:import_req, reftype:imptype );
+  if ( empty import_reference )
```

Comment:
Seems odd for a translation utility function to know so much about import references.  Perhaps this intelligence should be contained within ImportReference::get()?  Do we have (or envision) any situations in which the caller of that operation really wants an empty import reference returned to it?  If not, then it seems reasonable for the operation to return a non-empty one, creating a new import reference if one does not exist.